atom/browser/atom_browser_client.cc

```diff
@@ -970,6 +974,18 @@ bool AtomBrowserClient::WillCreateURLLoaderFactory(
   return true;
 }
 
+#if defined(OS_WIN)
+bool AtomBrowserClient::PreSpawnRenderer(sandbox::TargetPolicy* policy) {
+  // Allow crashpad to communicate via named pipe.
+  sandbox::ResultCode result = policy->AddRule(
+      sandbox::TargetPolicy::SUBSYS_FILES,
+      sandbox::TargetPolicy::FILES_ALLOW_ANY, L"\\??\\pipe\\crashpad_*");
```

Comment:
This confuses me too, in theory using pipes requires explicit allowance from sandbox, and it does in our case.

It is probably because Chrome organizes crashpad code into `chrome_elf.dll` and then access to the named pipe come from the same module instead of different processes. We can do the same in Electron but it would be another refactoring and should probably be done in a separate PR.

Reply:
Hmm, I'm not sure how `chrome_elf.dll` would play into that. If you load a dll in two separate processes, it's still two separate processes...

Does Chrome set up crashpad before the sandbox is engaged?

Reply:
It seems that they set it up before sandbox:
https://cs.chromium.org/chromium/src/chrome/app/chrome_main_delegate.cc?type=cs&q=v8_crashpad_support&sq=package:chromium&g=0&l=645

That should explain the difference on the behavior.